03 Writing Algorithms/28 Indicators/04 Automatic Indicators/99 Examples.html

```diff
@@ -0,0 +1,128 @@
+<p>The following examples demonstrate some common practices for using automatic indicators.</p>
+
+<h4>Example 1: Trade With Automatic Indicators</h4>
+<p>The following algorithm selects the top 20 liquid stocks. Then, we hold the stocks with price &gt; <a href="/docs/v2/writing-algorithms/indicators/supported-indicators/exponential-moving-average">EMA</a> &gt; <a href="/docs/v2/writing-algorithms/indicators/supported-indicators/simple-moving-average">SMA</a>, which indicates an upward accelerating trend, with aid of automatic indicators.</p>
+<div class="section-example-container">
+    <pre class="csharp">public class AutomaticIndicatorAlgorithm : QCAlgorithm
+{
+    private Dictionary&lt;Symbol, SymbolData&gt; _symbolData = new();
+
+    public override void Initialize()
+    {
+        SetStartDate(2021, 1, 1);
+        SetEndDate(2021, 2, 1);
+        
+        // Select the top 20 liquid stocks for popularity stocks to trade trend more efficiently.
+        AddUniverse(Universe.DollarVolume.Top(20));
+
+        // Set a scheduled event to rebalance daily on the daily indicator signals.
+        Schedule.On(
+            DateRules.EveryDay(),
+            TimeRules.At(9, 31),
+            Rebalance
+        );
+    }
+
+    private void Rebalance()
+    {
+        var symbolsToBuy = new List&lt;Symbol&gt;();
+        
+        foreach (var (symbol, symbolData) in _symbolData)
+        {
+            // Buy the stocks where price is above EMA and EMA is above SMA, meaning its trned is upward accelerating.
+            if (symbolData.IsReady &amp;&amp; Securities[symbol].Price &gt; symbolData.Ema &amp;&amp; symbolData.Ema &gt; symbolData.Sma)
+            {
+                symbolsToBuy.Add(symbol);
+            }
+        }
+
+        // Equally invest into the selected stocks to evenly dissipate the capital risk.
+        var count = symbolsToBuy.Count;
+        if (count &gt; 0)
+        {
+            var targets = symbolsToBuy.Select(symbol =&gt; new PortfolioTarget(symbol, 1m / count)).ToList();
+            // Liquidate the positions that are not in upward trend or not being popular anymore.
+            SetHoldings(targets, liquidateExistingHoldings: true);
+        }
+    }
+
+    public override void OnSecuritiesChanged(SecurityChanges changes)
+    {
+        foreach (var added in changes.AddedSecurities)
+        {
+            // Instantiate a new instance of SymbolData object to hold indicators to use.
+            _symbolData[added.Symbol] = new SymbolData(this, added.Symbol);
+        }
+
+        foreach (var removed in changes.RemovedSecurities)
+        {
+            // Remove the data subscription of indicators to release computation resources when leaving the universe.
+            _symbolData.Remove(removed.Symbol);
```

Comment:
`SymbolData` should implement a Dispose-like method and call `DiregisterIndicator`.